version: '3.7'
services:
    postgres:
        image: helsinkitest/postgis:9.6-2.5-alpine
        restart: on-failure
        environment:
            POSTGRES_USER: open_city_profile
            POSTGRES_PASSWORD: open_city_profile
            POSTGRES_DB: open_city_profile
        ports:
            - 5433:5432
        volumes:
          - pgdata:/var/lib/postgresql/data
        container_name: profile-db

    elk:
        image: sebp/elk
        restart: on-failure
        ports:
            - "5601:5601"
            - "9200:9200"
            - "5959:5959"
            - "5044:5044"
        container_name: profile-elk
        volumes:
            - ./elk/03-tcp-input.conf:/etc/logstash/conf.d/03-tcp-input.conf

    django:
        build:
            context: .
            target: development
        env_file:
            - docker-compose.env.yaml
        environment:
            DATABASE_URL: postgres://open_city_profile:open_city_profile@profile-db/open_city_profile
            DATABASE_HOST: "profile-db"
        volumes:
            - .:/app
            - django-media-volume:/var/media/
        ports:
            - "8080:8080"
        depends_on:
            - postgres
            - elk
        container_name: profile-backend

volumes:
        pgdata:
            driver: local
        django-media-volume:


networks:
    default:
        name: helsinki
